### ISSUE TITLE
zen_get_uprid functionality different admin/storefront

### ISSUE THREAD

**Issue by lat9** [zencart/zencart#1559]
What the heck!  I've been chasing my tail on this for a while now, to find that the processing of the subject function is **_significantly_** different in the admin implementation (functions/general.php) from that on the storefront (functions/functions_general.php).

Admin side:
```
  function zen_get_uprid($prid, $params) {
    $uprid = $prid;
    if ( (is_array($params)) && (!strstr($prid, '{')) ) {
      while (list($option, $value) = each($params)) {
        $uprid = $uprid . '{' . $option . '}' . $value;
      }
    }

    return $uprid;
  }
```
Storefront:
```
  function zen_get_uprid($prid, $params) {
//print_r($params);
    $uprid = $prid;
    if ( (is_array($params)) && (!strstr($prid, ':')) ) {
      while (list($option, $value) = each($params)) {
        if (is_array($value)) {
          while (list($opt, $val) = each($value)) {
            $uprid = $uprid . '{' . $option . '}' . trim($opt);
          }
        } else {
        //CLR 030714 Add processing around $value. This is needed for text attributes.
            $uprid = $uprid . '{' . $option . '}' . trim($value);
        }
      }      //CLR 030228 Add else stmt to process product ids passed in by other routines.
      $md_uprid = '';

      $md_uprid = md5($uprid);
      return $prid . ':' . $md_uprid;
    } else {
      return $prid;
    }
  }
```

I'd suggest "aligning" the admin version to reflect the storefront version and potentially modifying both to remove the use to the to-be-deprecated **each()**.

**Comment by drbyte** [comment]
Agreed. Making the Admin version of the function match the storefront version is a good idea. And after having looked at the usages of the function, I don't see any harm or backward-compatibility breaks coming from it. 

I'd accept a PR for this.

**Comment by lat9** [comment]
Will do.

**Comment by drbyte** [comment]
Closed by #1560
